### ISSUE TITLE
Feature Request - Cookie-functionality for WebViews

### ISSUE THREAD

**Issue by kasperrt** [xamarin/Xamarin.Forms#3262]
## Summary
Add easy, accessible functions for setting/removing/clearing cookies from a specified website in a WebView.

## API Changes

In order to add a cookie to a WebView instance, have a function in the lines of (This function can also function as editing a cookie, if there already exists a cookie with the specified cookie.Name. If an expiration date is set to a date in the past, the cookie will be deleted):

```
var webViewElement = new WebView();
webViewElement.AddCookie(Cookie cookieElement, Uri desiredDomain);
```

In order to get cookie(s) from a site

```
var webViewElement = new WebView();
webViewElement.GetCookie(string cookieName, Uri desiredDomain);
```

In order to clear cookie(s) from a site

```
var webViewElement = new WebView();
webViewElement.ClearCookies(Uri desiredDomain);
```

In order to get all cookies from a site

```
var webViewElement = new WebView();
webViewElement.GetAllCookies(Uri desiredDomain);
```

For security-purposes, the domain-name for the cookie might be desided internally, and use the current url/domain instead of having it specified as userinput.

## Intended Use Case

It could open for "seamless"-logins for WebViews, if, for example, one was to use an IdentityProvider to log into the app, display some data, and having the user navigate to the WebView. With the possibility of setting/adding/removing cookies "manually", the user could automatically be logged in, without having to be prompted again with authentication-screens.

## Difficulty: Easy

I'd think this should be an easy add, since cookiestorage and such already exists in the native elements, but that's just my assumption.

**Comment by warthenr** [comment]
Is there possibly a workaround for this capability?

**Comment by CliffAgius** [comment]
Is this still wanted as a feature?  I have just had to implement a custom renderer to achieve this taking a lead from a forum post:

[https://forums.xamarin.com/discussion/41039/setting-cookies-in-a-webview](url)

So as I have it working I was thinking of adding it to the WebView but didn't want to do this is it's not wanted. @samhouts @pauldipietro what do you think?

**Comment by warthenr** [comment]
I have a really bad hack using WinForms to integrate this in.  I would still love this built into the WebView itselv, but if you don't include it in, maybe you can post your code for the custom renderer somewhere for others to use.

**Comment by CliffAgius** [comment]
Use the link in my previous post I have basically taken the code from there and hacked it about to fit into my project.  I agree it would be nice to be in the XF WebView but until someone from the Xamarin team agree's it's ok to add there is no point doing the PR as this really is an enhancement which they may have reasons to turn down.

**Comment by samhouts** [comment]
@CliffAgius Go for it! Thanks!

**Comment by CliffAgius** [comment]
I have picked this one up to see if I can get it working.

**Comment by samhouts** [comment]
closed by #8169
